component/grant-type/src/main/java/org/wso2/carbon/identity/oauth2/grant/jwt/JWTBearerGrantHandler.java

```diff
@@ -165,8 +200,11 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
                 // if no IDPs were found for a given name, the IdentityProviderManager returns a dummy IDP with the
                 // name "default". We need to handle this case.
                 if (StringUtils.equalsIgnoreCase(identityProvider.getIdentityProviderName(), DEFAULT_IDP_NAME)) {
-                    log.error("Default IDP Found. This means a valid IDP was not found for issuer: " + jwtIssuer);
-                    handleException("No Registered IDP found for the JWT with issuer name : " + jwtIssuer);
+                    //if the id token is used as the assertion below code snippet will executed
```

Comment:
if the id token is used as the assertion below code snippet will executed --> check whether this jwt was issued by the resident identity provider